### PR TITLE
Update packageIndex.json with packages from release branches

### PR DESCRIPTION
**Pull request by joperezr**
Updating packageIndex to include stuff shipped in release/2.1 and release/2.2 branches.

cc: @ahsonkhan @ericstj 

@ahsonkhan @GrabYourPitchforks As FYI this change will make it such that all new packages built out of master (includding System.Text.Json and Encodings.Web) will get the latest version of System.Memroy which will have the fix for the binding redirect problem.

### CODE REVIEW

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json [hunk 1]
```diff
@@ -3167,7 +3207,7 @@
         "4.1.1.0": "4.1.1",
         "4.1.2.0": "4.3.0",
         "4.1.3.0": "4.4.0",
-        "4.1.4.0": "4.5.0",
```
Comment (ahsonkhan) [inline pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: **Question:** Is trailing commas not supported? And if so, what kind of issues could this have caused previously?

Comment (joperezr) [reply pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: Commas are supported, and this wasn’t an issue. I wrote a tool that generated these changes for me using an existing PackageIndex object model (which I’m pretty sure it uses Newtonsoft.json ) and this tool removed the coma if there was no additional item.

**Comment by ViktorHofer** [comment]
Should we also update harvestPackages.props? (+ in master)

**Comment by joperezr** [comment]
I need to ingest https://github.com/dotnet/arcade/pull/2798 in order to have this passing CI.

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json [hunk 2]
```diff
@@ -151,9 +152,14 @@
...
-      ],
-      "BaselineVersion": "2.0.0",
+        "2.1.0",
+        "2.1.1",
+        "2.1.2",
+        "2.1.3",
+        "2.1.4",
+        "2.2.0"
+      ],
+      "BaselineVersion": "2.2.0",
```
Comment (joperezr) [inline pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: @ericstj should I still update baseline version for Microsoft.NETCore.Platforms/Targets package as well? Does it even matter?

**Comment by joperezr** [comment]
I think this is now ready for review, @ericstj PTAL when you get the chance.

File: external/harvestPackages/harvestPackages.props [hunk 3]
```diff
@@ -62,10 +62,10 @@
   <Version>4.5.0</Version>
 </PackageReference>
 <PackageReference Include="System.Data.SqlClient">
-  <Version>4.5.1</Version>
+  <Version>4.6.0</Version>
```
Comment (ericstj) [inline external/harvestPackages/harvestPackages.props]: I think that SQLClient is also working on servicing release.

**Pull request merged**